python/tchannel/tornado/hyperbahn.py

```diff
@@ -59,31 +69,40 @@ def advertise(tchannel, service, routers):
         tchannel.peers.get(router)
 
     @tornado.gen.coroutine
-    def _register():
-        response = yield tchannel.request(service='hyperbahn').send(
-            arg1='ad',  # advertise
-            arg2='',
-            arg3=json.dumps({
-                'services': [
-                    {
-                        'serviceName': service,
-                        'cost': 0,
-                    }
-                ]
-            }),
-            headers={'as': 'json'},
-        )
-
-        if response.code not in ErrorCode:
-            # re-register every ``DEFAULT_TTL`` seconds
-            tornado.ioloop.IOLoop.current().call_later(
-                delay=DEFAULT_TTL,
-                callback=_register,
+    def _register(attempt_counter=0):
+
+        try:
+            response = yield tchannel.request(service='hyperbahn').send(
+                arg1='ad',  # advertise
+                arg2='',
+                arg3=json.dumps({
+                    'services': [
+                        {
+                            'serviceName': service,
+                            'cost': 0,
+                        }
+                    ]
+                }),
+                headers={'as': 'json'},
+                attempt_times=1,
             )
-            raise tornado.gen.Return(response)
+        except Exception as e:
```

Comment:
The exception caught in this level is for backoff approach. Advertise method will always exist but run in certain period.